### DISCUSSION TITLE
Splink Tests?

### DISCUSSION THREAD

**Discussion by LeCodeMinister** [moj-analytical-services/splink#2475] [General]
Hi there!

I've used Splink extensively over the past few months for work and it's been amazing. However, I was wondering if there's a recommended way to test any Splink code?

There's a requirement at my workplace to hit at least 80% coverage and with the code for model training/predicting & clustering being in two different files in my case, it seems doable. But still, I wanted to hear how people with similar requirements have gone about this?

Cheers!

**Comment by nabebaye** [comment]
I approached this problem by using a small data set, running Splink on it, and then confirming expected results. I then use the same inputs in an integration test and confirm that the outputs are the same to avoid regression and other drift in results due to code around it changing
